analytical_engine/apps/assortativity/degree/degree_assortativity_coefficient_context.h

```diff
@@ -0,0 +1,85 @@
+/** Copyright 2020 Alibaba Group Holding Limited.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef ANALYTICAL_ENGINE_APPS_ASSORTATIVITY_DEGREE_DEGREE_ASSORTATIVITY_COEFFICIENT_CONTEXT_H_
+#define ANALYTICAL_ENGINE_APPS_ASSORTATIVITY_DEGREE_DEGREE_ASSORTATIVITY_COEFFICIENT_CONTEXT_H_
+
+#include <limits>
+#include <string>
+#include <unordered_map>
+#include <utility>
+
+#include "grape/grape.h"
+
+#include "apps/assortativity/utils.h"
+#include "core/app/app_base.h"
+#include "core/context/tensor_context.h"
+
+namespace gs {
+
+template <typename FRAG_T>
+class DegreeAssortativityContext : public TensorContext<FRAG_T, double> {
+ public:
+  // using vid_t = typename FRAG_T::vid_t;
+  explicit DegreeAssortativityContext(const FRAG_T& fragment)
+      : TensorContext<FRAG_T, double>(fragment) {}
+  typedef double degree_t;
+  void Init(grape::DefaultMessageManager& messages,
+            std::string source_degree_type = "out",
+            std::string target_degree_type = "in", bool directed = false,
+            bool weighted = false) {
+    merge_stage = false;
+    this->directed = directed;
```

Comment:
use frag->directed() to check direction of graph. no need to add a directed variable in context

Reply:
ImmutableEdgecutFragment doesn't have directed() function

Reply:
if we use the application in GraphScope, we can just focus on `ArrowProjectedFragment` and `DynamicProjectedFragment`, these fragment has `directed()` method.